src/Modules/OrchardCore.Commerce.Payment/Controllers/PaymentController.cs

```diff
@@ -88,14 +88,14 @@ await this.SafeJsonAsync(async () =>
             };
         });
 
-    [Route("checkout/validate/{providerName}")]
+    [Route("checkout/validate/{providerName}/{paymentIntentId}")]
```

Comment:
Rename the parameter to `paymentId` as well. Also make it optional by appending `?` to the template like this:
`[Route("checkout/validate/{providerName}/{paymentId?}")]`